Civi/Api4/Service/Schema/SchemaMapBuilder.php

```diff
@@ -126,7 +126,13 @@ private function addCustomFields(SchemaMap $map, Table $baseTable, string $entit
       // Add joins for entityReference fields
       foreach ($customGroup['fields'] as $field) {
         if ($field['data_type'] === 'EntityReference' && isset($field['fk_entity'])) {
-          $targetTable = self::getTableName($field['fk_entity']);
+          $targetEntity = $field['fk_entity'];
+          if (!isset(AllCoreTables::getEntities()[$targetEntity])) {
```

Comment:
Since we're in the API space let's do
```suggestion
          if (!CoreUtil::getTableName($targetEntity)) {
```

Reply:
Actually no, that might cause a loop mightn't it?

Reply:
Let's do
```suggestion
          if (!self::getTableName($targetEntity)) {
```